components/ws-proxy/pkg/proxy/routes.go

```diff
@@ -436,63 +442,72 @@ func installWorkspacePortRoutes(r *mux.Router, config *RouteHandlerConfig, infoP
 				withXFrameOptionsFilter(),
 			)(rw, r)
 		},
-	)
+	).Name("root")
 
 	return nil
 }
 
 // workspacePodResolver resolves to the workspace pod's url from the given request.
-func workspacePodResolver(config *Config, infoProvider WorkspaceInfoProvider, req *http.Request) (url *url.URL, err error) {
+func workspacePodResolver(config *Config, infoProvider WorkspaceInfoProvider, req *http.Request) (url *url.URL, resource string, err error) {
 	coords := getWorkspaceCoords(req)
 	var port string
 	if coords.Debug {
+		resource = "debug_workspace"
 		port = fmt.Sprint(config.WorkspacePodConfig.IDEDebugPort)
 	} else {
+		resource = "workspace"
 		port = fmt.Sprint(config.WorkspacePodConfig.TheiaPort)
 	}
 	workspaceInfo := infoProvider.WorkspaceInfo(coords.ID)
-	return buildWorkspacePodURL(workspaceInfo.IPAddress, port)
+	url, err = buildWorkspacePodURL(workspaceInfo.IPAddress, port)
+	return
 }
 
 // workspacePodPortResolver resolves to the workspace pods ports.
-func workspacePodPortResolver(config *Config, infoProvider WorkspaceInfoProvider, req *http.Request) (url *url.URL, err error) {
+func workspacePodPortResolver(config *Config, infoProvider WorkspaceInfoProvider, req *http.Request) (url *url.URL, resource string, err error) {
 	coords := getWorkspaceCoords(req)
 	workspaceInfo := infoProvider.WorkspaceInfo(coords.ID)
 	var port string
 	if coords.Debug {
+		resource = "debug_workspace_port"
 		port = fmt.Sprint(config.WorkspacePodConfig.DebugWorkspaceProxyPort)
 	} else {
+		resource = "workspace_port"
```

Comment:
Consider excluding workspace ports, as they mainly involve user apps and may result in internal errors. However, retaining them could be useful, for example, if there's no status code (0), it may indicate ws-proxy connectivity issues, suggesting potential problems on our end.

If filtering them costly, let's filter in the code. Otherwise, implement filtering at the query level later?